### PR TITLE
Add crc calculation in loading module

### PR DESCRIPTION
**Pull request by sum2012**
Record in mpeg and psmfPlayer

### CODE REVIEW

**Comment by sum2012** [comment]
I have #include "UI/GameScreen.h"
Why I get this error ?
jni/../../Core/HLE/sceKernelModule.cpp:1150: error: undefined reference to 'int2hexstr(int)'

**Comment by hrydgard** [comment]
Why do you want to turn the CRC into a string, and not just keep it as an integer? Also that last change isn't gonna work, you can't return a pointer to a local variable in the function.

**Comment by sum2012** [comment]
CRC into a string is for debug only.
I checked with other Hashtool that the CRC is right.
I will undo the string later.
I need to go out now

File: Core/HLE/sceKernelModule.cpp [hunk 1]
```diff
@@ -1144,6 +1144,8 @@ static int gzipDecompress(u8 *OutBuffer, int OutBufferLength, u8 *InBuffer) {
 }
 
 static PSPModule *__KernelLoadELFFromPtr(const u8 *ptr, size_t elfSize, u32 loadAddress, bool fromTop, std::string *error_string, u32 *magic, u32 &error) {
+	u32 crc = crc32(0, Z_NULL, 0);
+	crc = crc32(crc, ptr, elfSize);
```
[the remark this answers is not in the document]
Comment (sum2012) [reply Core/HLE/sceKernelModule.cpp:1148]: I copied the code from
https://github.com/hrydgard/ppsspp/blob/master/Core/FileSystems/BlockDevices.cpp#L57
Now I tested , your suggestion is okay

[the remark this answers is not in the document]
Comment (hrydgard) [reply Core/HLE/sceKernelModule.cpp]: Hm, I see. Yeah that one makes sense when you then loop over it, in this case one call is neater I think. Thanks!

**Review by hrydgard — APPROVED** [review]

**Pull request merged**